File: LeetCode_Marathons/LCM4/Medium/Beautiful_Arrangement_II/a.go
func constructArray(n int, k int) []int {
    l, r := 1, k + 1
    ans := []int{}
    for l <= r {
        ans = append(ans, l)
        l++
        if l > r {
            break
        }
        ans = append(ans, r)
        r--
    }
    for i := k + 2; i <= n; i++ {
        ans = append(ans, i)
    }
    return ans
}

File: LeetCode_Marathons/LCM4/Medium/Beautiful_Arrangement_II/a.cpp
class Solution {
public:
    vector<int> constructArray(int n, int k) {
        vector<int> ans;
        int l = 1, r = k + 1;
        while (l <= r) {
            ans.push_back(l++);
            if (l > r) break;
            ans.push_back(r--);
        }
        for (int i = k + 2; i <= n; i++) ans.push_back(i);
        return ans;
    }
};